src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,229 @@
+    // Build the table reference
+    TableId tableRef = TableId.of(projectId, dataset, tableName);
+
+    // Get the table schema
+    Schema schema = bigQuery.getTable(tableRef).getDefinition().getSchema();
+
+    // Iterate over the fields and print the column name and type
+    int currentColumnCount = 1;
+    while (currentColumnCount <= columnCountSource) {
+      for (Field field : schema.getFields()) {
+        String columnName = field.getName();
+        String columnType = field.getType().toString();
+        String columnTypeName = field.getType().getStandardType().name();
+        System.out.println("Column Name: " + columnName);
+        System.out.println("Column Type: " + columnType);
+        System.out.println("Column TypeName: " + columnTypeName);
+
+        switch (columnType) {
+
+          case "BOOLEAN":
+            boolean sfDateString = salesforceData.get(0).get(columnName).getAsBoolean();
+            boolean bqDateString = bigQueryData.get(jsonObjectIdx).get(columnName).getAsBoolean();
+            Assert.assertEquals(sfDateString, bqDateString);
+            break;
+
+          case "FLOAT":
+            Double source = salesforceData.get(0).get(columnName).getAsDouble();
+            Double target = bigQueryData.get(jsonObjectIdx).get(columnName).getAsDouble();
+            Assert.assertTrue(String.format("Different values found for column: %s", columnName),
+                              source.compareTo(target) == 0);
+
+            break;
+
+          case "TIMESTAMP":
+            OffsetDateTime sourceDateTime = OffsetDateTime.parse(
+              salesforceData.get(0).get(columnName).getAsString(),
+              DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ"));
+            // Format the source timestamp in the desired format
+            String formattedSourceTimestamp = sourceDateTime.toString();
+            String formattedTargetTimestamp = bigQueryData.get(jsonObjectIdx).get(columnName).getAsString();
+            Assert.assertEquals(formattedTargetTimestamp, formattedSourceTimestamp);
+            break;
+
+          case "TIME":
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss.SSSX");
+            DateTimeFormatter formatter2 = DateTimeFormatter.ofPattern("HH:mm:ss");
+            // Parse the timestamp
+            LocalTime sourceTime = LocalTime.parse(salesforceData.get(jsonObjectIdx).get(columnName).getAsString(),
+                                              formatter);
+            LocalTime TargetTime = LocalTime.parse(bigQueryData.get(jsonObjectIdx).get(columnName).getAsString(),
+                                              formatter2);
+            Assert.assertEquals(sourceTime, TargetTime);
+            break;
+
+          case "DATE":
+            JsonElement jsonElem1 = salesforceData.get(0).get(columnName);
+            Date sourceVal = (jsonElem1 != null && !jsonElem1.isJsonNull()) ? Date.valueOf(
```

Comment:
Why do we need to check null check here?

Reply:
Because some of the fields are null in Date by default from salesforce.